From 9202afdc7a347e4b8112da4a0cf8a860a966f96b Mon Sep 17 00:00:00 2001
From: Raphael Ludwig <raphael.ludwig@happywetter.at>
Date: Thu, 9 Sep 2021 12:10:31 +0200
Subject: [PATCH] 3408: Fix: ARM setStackMode() parameter

Fix: setStackMode now uses Rn0003 as parameter instead of calling isUsingMainStack
---
 Ghidra/Processors/ARM/data/languages/ARMTHUMBinstructions.sinc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Ghidra/Processors/ARM/data/languages/ARMTHUMBinstructions.sinc b/Ghidra/Processors/ARM/data/languages/ARMTHUMBinstructions.sinc
index 4ba138801..a4e54e1f7 100644
--- a/Ghidra/Processors/ARM/data/languages/ARMTHUMBinstructions.sinc
+++ b/Ghidra/Processors/ARM/data/languages/ARMTHUMBinstructions.sinc
@@ -2843,7 +2843,7 @@ define pcodeop setStackMode;
 # TODO: not sure about the following semantics  
   b = isThreadMode();
   if (!b) goto inst_next;
-  stackMode:1 = isUsingMainStack() == 1:1;
+  stackMode:1 = (Rn0003 & 2) != 0;
   setStackMode(stackMode);
 # TODO: should we set sp ?
 }
-- 
2.38.1

